infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -495,12 +485,20 @@ public void processLockRelease() {
         stopLogReplication();
         isLeader.set(false);
         // Signal Log Replication Server/Sink to stop receiving messages, leadership loss
-        if (topologyDescriptor.getLocalClusterDescriptor() != null && topologyDescriptor.getLocalClusterDescriptor().getRole() == ClusterRole.SINK) {
+        if (isSink(topologyDescriptor)) {
             interClusterServerNode.setLeadership(false);
         }
         recordLockRelease();
     }
 
+    private boolean isSource(TopologyDescriptor topology) {
+        return !topology.getRemoteSinkClusters().isEmpty();
+    }
+
+    private boolean isSink(TopologyDescriptor topology) {
+        return !topology.getRemoteSourceClusters().isEmpty();
+    }
```

Comment:
I had to do this because in `onSinkClusterAddRemove()` the new topology was set much later.
But looking with fresh eyes today, the statements in `onSinkClusterAddRemove()` needs to be reordered. 

I'll have a fix in the next push.